### ISSUE TITLE
Error in firebase_functions/private/util.py processing event time attribute

### ISSUE THREAD

**Issue by ekalosak** [firebase/firebase-functions-python#136]
After updating firebase-tools, using the functions emulator, any time an `@firestore_fn`-decorated function is executed, I get this error:
```
...
>    File "/.../venv/lib/python3.11/site-packages/firebase_functions/firestore_fn.py", line 115, in _firestore_endpoint_handler
>      is_nanoseconds = _util.is_precision_timestamp(time)
>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>    File "/.../venv/lib/python3.11/site-packages/firebase_functions/private/util.py", line 333, in is_precision_timestamp
>      _, s_fraction = time.split(".")
>      ^^^^^^^^^^^^^
>  ValueError: not enough values to unpack (expected 2, got 1)
```

The relevant code includes [this block from firebase_fn.py](https://github.com/firebase/firebase-functions-python/blob/main/src/firebase_functions/firestore_fn.py#L86-L115) and [this block from private/util.py](https://github.com/firebase/firebase-functions-python/blob/main/src/firebase_functions/private/util.py#L340-L343).

After upgrading to the latest firebase tools `npm install -g firebase-tools` to `v12.5.4`, the CloudEvent event's time attribute has this iso format: `2023-09-16T13:32:06Z` extracted by adding a print() call to the firebase_functions' site-package source.

So, that unprotected `time.split('.')` call isn't accounting for the ostensible change to the time attribute's format.

---
I'm using Python 3.11.4 on macOS 12.6 with firebase-functions v0.1.1 (latest).

**Comment by ekalosak** [comment]
It's worth noting that downgrading `firebase-tools` from the latest `v12.5.4` to the previous `v12.5.3` fixes the issue.

tl;dr fix this bug by running: `npm install -g firebase-tools@12.5.4`

Firebase staff - please consider adding an integration test that covers `@firebase_fn`-triggered functions to the firebase-tools CI.

**Comment by mitch-36** [comment]
Thanks for sharing. 
`npm install -g firebase-tools@12.5.3`
Also worked for me.

**Comment by joaqo** [comment]
I have the same issue, but running `npm install -g firebase-tools@12.5.3` didn't fix it for me. What command is invoking `firebase-tools`? I am assuming its used by `firebase emulators:start`?

**Comment by brianellin** [comment]
Same issue for me using the latest firebase.tools. Using 12.5.3 via npm for now until the issue is resolved. 

Also, I'd recommend some better docs on "installing" a specific version binary from the firebase.tools website.

**Comment by exaby73** [comment]
Snce this issue has been fixed, I'll close this issue. Please comment if you disagree with my decision

**Comment by jackall3n** [comment]
> Snce this issue has been fixed, I'll close this issue. Please comment if you disagree with my decision

This is still an issue in the latest version, this issue shouldn't be closed.

**Comment by joaqo** [comment]
Yeah, this is the 2nd recent issue @exaby73 has closed that hasn't actually been fixed.

**Comment by exaby73** [comment]
@jackall3n @joaqo Sorry about this. I realise that the workaround was downgrading the firebase_tools version. Correct me if I am wrong. I'll reopen this issue and investigate further

**Comment by antopolskiy** [comment]
+1 this issue still happens and if not for this thread it would have been very difficult for me to resolve. I spent ~1 hour fruitlessly playing around with version of various components before finding this. `npm install -g firebase-tools@12.5.3` works.

**Comment by zsais** [comment]
+1 this issue still exists

**Comment by exaby73** [comment]
Could you please check with the latest version `0.2.0` and confirm if this issue is fixed on your side? There was a PR that was merged but not released earlier which may have fixed this issue

**Comment by antopolskiy** [comment]
I just tried it with firebase-tools v13.3.0 and it still fails.

**Comment by exaby73** [comment]
Hey @antopolskiy if you can still reproduce it, could you provide me with your stack trace?

**Comment by CorieW** [comment]
Hi @ekalosak.

Thanks for reporting this issue. I've managed to reproduce ([MRE](https://github.com/invertase/cloud-team-mre/tree/firebase/firebase-functions-python/issue-136/firebase/firebase-functions-python/issue-136)) it using the following versions:

- **firebase_functions:** v0.1.1
- **firebase-tools:** v12.5.4 (and all later versions)
- **firebase-admin:** v6.2.0

To fix this issue, you can either:

- Update from **firebase_functions** v0.1.1 to v0.1.2.
- Update to the latest version of **firebase_functions** (v0.4.3).

Since there are good solutions to this issue, I'm going to close this issue here. However, if you have any further comments or problems with the solutions, let me know.
